public/app/core/components/NestedFolderPicker/useFoldersQueryAppPlatform.ts

```diff
@@ -145,6 +148,14 @@ export function useFoldersQueryAppPlatform({
 
         const childResponse = folderIsOpen && state.responseByParent[name];
         if (childResponse) {
+          // If we finished loading and there are no children add to empty list
+          if (
+            childResponse.data &&
+            childResponse.status !== QueryStatus.pending &&
+            childResponse.data.hits.length === 0
+          ) {
+            setEmptyFolders((prev) => new Set(prev).add(name));
```

Comment:
Do we need the Set constructor here as the default state value is already a set?

Reply:
This is to copy the Set and have an immutability for the state.